### PR TITLE
Wrap ComponentContext with RouterContext

### PR DESCRIPTION
**Pull request by arkivanov**
Please choose either this PR or #58.

### CODE REVIEW

File: decompose-router/src/commonMain/kotlin/io/github/xxfast/decompose/router/Router.kt [hunk 1]
```diff
@@ -51,16 +53,31 @@ fun <C : Parcelable> rememberRouter(
...
+      Router(
+        navigator = navigation,
+        stack = routerContext.childStack(
+          source = navigation,
+          initialStack = { stack },
+          configurationClass = type,
+          key = keyStr,
+          handleBackButton = handleBackButton,
+          childFactory = { _, childComponentContext -> RouterContext(childComponentContext) },
+        ).asState(routerContext.lifecycle),
```
Comment (xxfast) [inline decompose-router/src/commonMain/kotlin/io/github/xxfast/decompose/router/Router.kt:71]: I might lift this out to a variable for clarity 
```kotlin
val navigation = StackNavigation<C>()
val stack: State<ChildStack<C, RouterContext>> = routerContext
  .childStack(
    source = navigation,
    initialStack = { stack },
    configurationClass = type,
    key = keyStr,
    handleBackButton = handleBackButton,
    childFactory = { _, childComponentContext -> RouterContext(childComponentContext) },
  )
  .asState(routerContext.lifecycle)

Router(navigation, stack)
```

**Review by xxfast — APPROVED** [review]
Nice work 🙌  Thank you very much for the contribution

File: decompose-router/src/commonMain/kotlin/io/github/xxfast/decompose/router/Router.kt [hunk 2]
```diff
@@ -51,16 +53,31 @@ fun <C : Parcelable> rememberRouter(
...
-    initialStack = { stack },
-    key = key.toString(), // Has to use strings for Android 😢
-    handleBackButton = handleBackButton,
-    type = type,
-  )
+  val routerContext = LocalRouterContext.current
+  val keyStr = key.toString()
 
-  return remember { Router(navigator = navigator, stack = childStackState) }
+  return remember {
```
Comment (xxfast) [inline decompose-router/src/commonMain/kotlin/io/github/xxfast/decompose/router/Router.kt:59]: @arkivanov Do you think we need to key this remember with our  `key`
```kotlin
return remember(key) {
    routerContext.getOrCreate(key = keyStr) { .. }
}
```

Comment (arkivanov) [reply decompose-router/src/commonMain/kotlin/io/github/xxfast/decompose/router/Router.kt:59]: Yeah, you can use keyStr, the same as for getOrCreate

**Pull request merged**